src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Features/Blocks/disabled_feature_warning.html.twig

```diff
@@ -0,0 +1,32 @@
+{#**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *#}
+
+<div class="alert alert-warning" role="alert">
+  <p>
+    {{ 'This feature has been disabled. You can activate it here: %url%.'|trans({
+      '%url%': '<a href="' ~ path('admin_performance') ~ '#optional_features">' ~ 'Performance'|trans({}, 'Admin.Global') ~ '</a>'
```

Comment:
I'm not a fan of all this concatenation happening in the twig file, looks a bit like black magic
Maybe we should do this on controller side, or create a Twig extension for it ?

Reply:
I don't think we can make it clean from controller side, as there is this `<a>` tag being replaced, but maybe a twig extension would work here :thinking:

Reply:
after thinking about it, does it really make sense to have a twig extension which would simply render an `<a>` link? seems a bit overengineered to me :thinking:

Reply:
@matks, would this be considered less magical:
```twig
{% set href = path('admin_performance') ~ '#optional_features' %}
{% set linkText = 'Performance'|trans({}, 'Admin.Global') %}

<div class="alert alert-warning" role="alert">
  <p>
    {{ 'This feature has been disabled. You can activate it here: %url%.'|trans({
      '%url%': "<a href=\"#{href}\">#{linkText}</a>"
    }, 'Admin.Catalog.Notification')|raw }}
  </p>
</div>
```
:male_detective:

Reply:
@rokaszygmantas Yes ! It's so much better this way 😄